packages/react/src/hooks.ts

```diff
@@ -0,0 +1,30 @@
+import { parser } from "@chainsafe/eth-sso-consumer";
+import { useEffect } from "react";
+
+export function useAuthCallback(
+  searchParams = new URLSearchParams(window.location.search),
+): void {
+  useEffect(() => {
+    const auth = parser.parseAuth(searchParams);
+    if (auth) {
+      if (window.opener || "postMessage" in window.opener)
+        (window.opener as typeof window.parent).postMessage(auth.clone());
+      else window.parent.postMessage(auth.clone());
+    }
+  }, [searchParams]);
+}
+
+export function useSendTransactionCallback(
+  searchParams = new URLSearchParams(window.location.search),
+): void {
+  useEffect(() => {
+    const transaction = parser.parseSendTransaction(searchParams);
+    if (transaction) {
+      if (window.opener || "postMessage" in window.opener)
+        (window.opener as typeof window.parent).postMessage(
+          transaction.clone(),
+        );
+      else window.parent.postMessage(transaction.clone());
```

Comment:
maybe we can add into "consumer"/sdk method called "init" with instructions that in needs to be executed on page load which would listen for postMessages?

Reply:
I do not understand where the is confusion? Why is there a need for init on the callback route that notifies the react part that the response is received?